dev/archery/archery/linking.py

```diff
@@ -61,9 +63,84 @@ def list_dependency_names(self):
             names.append(name)
         return names
 
+    def _remove_weak_symbols(self, symbol_info):
+        return [line for line in symbol_info if not line.endswith(
+            (" v", " V", " w", " W"))]
+
+    def _remove_symbol_versions(self, symbol_info):
+        return [line.split('@')[0].strip() for line in symbol_info]
+
+    def list_symbols_for_dependency(self, dependency, remove_symbol_versions=False):
```

Comment:
Rather than take `dependency` as an argument, why not just access `self.path`?

Reply:
Same question for the other methods below?

Reply:
It can be done to the `list_undefined_symbols_for_dependency` function, though for this function I am not sure @pitrou 
I just printed the following 

### Code Diff

```python
def list_symbols_for_dependency(self, dependency, remove_symbol_versions=False):
        print("list_symbols_for_dependency: ", dependency, self.path)
        ...
```

### Output

```bash
list_symbols_for_dependency:  linux-vdso.so.1 /home/vibhatha/Documents/Work/Apache_Arrow/issues/gh-40018/arrow-dataset-15.0.2/x86_64/libarrow_dataset_jni.so
list_symbols_for_dependency:  /lib/x86_64-linux-gnu/librt.so.1 /home/vibhatha/Documents/Work/Apache_Arrow/issues/gh-40018/arrow-dataset-15.0.2/x86_64/libarrow_dataset_jni.so
list_symbols_for_dependency:  /lib/x86_64-linux-gnu/libpthread.so.0 /home/vibhatha/Documents/Work/Apache_Arrow/issues/gh-40018/arrow-dataset-15.0.2/x86_64/libarrow_dataset_jni.so
list_symbols_for_dependency:  /lib/x86_64-linux-gnu/libdl.so.2 /home/vibhatha/Documents/Work/Apache_Arrow/issues/gh-40018/arrow-dataset-15.0.2/x86_64/libarrow_dataset_jni.so
list_symbols_for_dependency:  /lib/x86_64-linux-gnu/libstdc++.so.6 /home/vibhatha/Documents/Work/Apache_Arrow/issues/gh-40018/arrow-dataset-15.0.2/x86_64/libarrow_dataset_jni.so
list_symbols_for_dependency:  /lib/x86_64-linux-gnu/libm.so.6 /home/vibhatha/Documents/Work/Apache_Arrow/issues/gh-40018/arrow-dataset-15.0.2/x86_64/libarrow_dataset_jni.so
list_symbols_for_dependency:  /lib/x86_64-linux-gnu/libgcc_s.so.1 /home/vibhatha/Documents/Work/Apache_Arrow/issues/gh-40018/arrow-dataset-15.0.2/x86_64/libarrow_dataset_jni.so
list_symbols_for_dependency:  /lib/x86_64-linux-gnu/libc.so.6 /home/vibhatha/Documents/Work/Apache_Arrow/issues/gh-40018/arrow-dataset-15.0.2/x86_64/libarrow_dataset_jni.so
list_symbols_for_dependency:  /lib64/ld-linux-x86-64.so.2 /home/vibhatha/Documents/Work/Apache_Arrow/issues/gh-40018/arrow-dataset-15.0.2/x86_64/libarrow_dataset_jni.so
```

Reply:
It is true that we use `self.path` to get all the other paths and get the libraries path, we could fold everything into just one function and remove this function if required. But my intention was to keep a util function which can list dependencies for a particular library path. Am I wrong here? Can we do better?